### PR TITLE
Use toml as configuration file format instead of json

### PR DESCRIPTION
**Pull request by penguwin**
This changes our default configuration format from **json** to **toml** as toml is a bit easier on the eye. It's also a **breaking change** regarding the configuration system as knoxite won't be able to read config files written in json after applying this. 

- [x] Update relevant documentation

### CODE REVIEW

**Comment by muesli** [comment]
We need to add a config file version const, and implement a migration function, which loads the old JSON config (`v0`) and converts it to a toml config file (`v1`).

File: cmd/knoxite/config/testdata/knoxite.conf [hunk 1]
```diff
@@ -1,20 +1,8 @@
...
-            ],
-            "restore_excludes": [
-                "just",
-                "an",
-                "example"
-            ]
-        }
-    }
-}
+[repositories]
```
Comment (muesli) [inline cmd/knoxite/config/testdata/knoxite.conf:1]: After rebasing master, don't forget to add the `pedantic` setting here.

**Comment by penguwin** [comment]
Ready for review

**Review by craftamap — DISMISSED** [review]
Looks good!

**Review by ma-hartma — DISMISSED** [review]
Great PR and a real improvement of the usability! It's much easier to read!

**Review by muesli — APPROVED** [review]
Sweet, LGTM!

**Pull request merged**